### PR TITLE
e2e: refactored ctlv3_make_mirror_test

### PR DESCRIPTION
**Pull request by sinsharat**
Modified the test cases to remove code duplicity and to make it more orgainised,

### CODE REVIEW

File: e2e/ctl_v3_make_mirror_test.go [hunk 1]
```diff
@@ -25,106 +25,31 @@ func TestCtlV3MakeMirrorModifyDestPrefix(t *testing.T) { testCtl(t, makeMirrorMo
...
 func makeMirrorNoDestPrefixTest(cx ctlCtx) {
+	flags := []string{"--prefix", "o_", "--no-dest-prefix"}
+	var kvs = []kv{{"o_key1", "val1"}, {"o_key2", "val2"}, {"o_key3", "val3"}}
+	var kvs2 = []kv{{"key1", "val1"}, {"key2", "val2"}, {"key3", "val3"}}
+	var srcprefix = "o_"
+	var destprefix = "key"
+	performMakeMirrorOp(cx, flags, kvs, kvs2, srcprefix, destprefix)
+}
+
+func performMakeMirrorOp(cx ctlCtx, flags []string, sourcekvs, destkvs []kv, srcprefix, destprefix string) {
```
Comment (gyuho) [inline e2e/ctl_v3_make_mirror_test.go]: `s/performMakeMirrorOp/testMakeMirrorOp/`?

File: e2e/ctl_v3_make_mirror_test.go [hunk 2]
```diff
@@ -147,8 +72,11 @@ func makeMirrorNoDestPrefixTest(cx ctlCtx) {
 		}
 	}()
 
-	cmdArgs := append(cx.PrefixArgs(), "make-mirror", "--prefix", "o_", "--no-dest-prefix", fmt.Sprintf("localhost:%d", mirrorcfg.basePort))
+	cmdArgs := append(cx.PrefixArgs(), "make-mirror")
+	cmdArgs = append(cmdArgs, flags...)
+	cmdArgs = append(cmdArgs, fmt.Sprintf("localhost:%d", mirrorcfg.basePort))
 	proc, err := spawnCmd(cmdArgs)
+
```
Comment (gyuho) [inline e2e/ctl_v3_make_mirror_test.go]: remove this blank line

Comment (sinsharat) [inline e2e/ctl_v3_make_mirror_test.go]: sure will remove it.

File: e2e/ctl_v3_make_mirror_test.go (hunk 1 above)
Comment (sinsharat) [inline e2e/ctl_v3_make_mirror_test.go]: sure will rename performMakeMirrorOp to testMakeMirrorOp

**Comment by sinsharat** [comment]
@gyuho I have done the changes as per your suggestion.
Thank you!

**Comment by gyuho** [comment]
lgtm. defer to @xiang90 

Thanks @sinsharat

**Comment by sinsharat** [comment]
@gyuho you are welcome :-)

File: e2e/ctl_v3_make_mirror_test.go [hunk 3]
```diff
@@ -25,106 +25,31 @@ func TestCtlV3MakeMirrorModifyDestPrefix(t *testing.T) { testCtl(t, makeMirrorMo
...
-		cx.t.Fatal(err)
-	}
+	var srcprefix = "o_"
+	var destprefix = "d_"
+	testMirrorOp(cx, flags, kvs, kvs2, srcprefix, destprefix)
 }
 
 func makeMirrorNoDestPrefixTest(cx ctlCtx) {
+	flags := []string{"--prefix", "o_", "--no-dest-prefix"}
+	var kvs = []kv{{"o_key1", "val1"}, {"o_key2", "val2"}, {"o_key3", "val3"}}
```
Comment (xiang90) [inline e2e/ctl_v3_make_mirror_test.go]: i prefer

```go
var (
...
)

```

File: e2e/ctl_v3_make_mirror_test.go [hunk 4]
```diff
@@ -25,106 +25,31 @@ func TestCtlV3MakeMirrorModifyDestPrefix(t *testing.T) { testCtl(t, makeMirrorMo
...
+	var kvs2 = []kv{{"key1", "val1"}, {"key2", "val2"}, {"key3", "val3"}}
+	var srcprefix = "o_"
+	var destprefix = "key"
+	testMirrorOp(cx, flags, kvs, kvs2, srcprefix, destprefix)
+}
+
+func testMirrorOp(cx ctlCtx, flags []string, sourcekvs, destkvs []kv, srcprefix, destprefix string) {
```
Comment (xiang90) [inline e2e/ctl_v3_make_mirror_test.go]: testMirrorCommand?

Comment (sinsharat) [inline e2e/ctl_v3_make_mirror_test.go]: @xiang90 sure will modify the name.

File: e2e/ctl_v3_make_mirror_test.go (hunk 3 above)
Comment (sinsharat) [inline e2e/ctl_v3_make_mirror_test.go]: @xiang90 sure will do the changes. Just one confirmation required, you want me to move all the variables under var block. right? and you want this to be done for all the functions right?

Comment (xiang90) [inline e2e/ctl_v3_make_mirror_test.go]: yes. i prefer so.

Comment (sinsharat) [inline e2e/ctl_v3_make_mirror_test.go]: @xiang90 sure will do the changes and let you know.
Thank you!

**Comment by sinsharat** [comment]
@xiang90 I have done the changes as per your suggestion.
Thank you!

**Comment by xiang90** [comment]
lgtm

**Pull request merged**